FHEM/14_BresserTemeo.pm

```diff
@@ -0,0 +1,405 @@
+##############################################
+# $Id$
+# The file is taken from the SIGNALduino project
+# see http://www.fhemwiki.de/wiki/SIGNALduino
+# and was modified by a few additions
+# to support BresserTemeo Sensors
+# S. Butzek & HJGode & Ralf9 2015-2016
+# B. Plutka 2016
+
+package main;
+
+use strict;
+use warnings;
+use POSIX;
+
+#use Data::Dumper;
+
+#####################################
+sub
+BresserTemeo_Initialize($)
+{
+  my ($hash) = @_;
+
+
+  $hash->{Match}     = "^u(44|51)#[A-F0-9]{17,19}";   # Laenge (Anzahl nibbles nach 0x75 )noch genauer spezifizieren
+  $hash->{DefFn}     = "BresserTemeo_Define";
+  $hash->{UndefFn}   = "BresserTemeo_Undef";
+  $hash->{AttrFn}    = "BresserTemeo_Attr";
+  $hash->{ParseFn}   = "BresserTemeo_Parse";
+  $hash->{AttrList}  = "IODev do_not_notify:0,1 showtime:0,1 "
+                       ."ignore:0,1 "
+                      ." $readingFnAttributes";
+                      
+  $hash->{AutoCreate}=
+        { "BresserTemeo.*" => { ATTR => "event-min-interval:.*:300 event-on-change-reading:.*", FILTER => "%NAME", GPLOT => "temp4hum4:Temp/Hum,", autocreateThreshold => "2:180"} };
+
+}
+
+
+#####################################
+sub
+BresserTemeo_Define($$)
+{
+  my ($hash, $def) = @_;
+  my @a = split("[ \t][ \t]*", $def);
+
+  return "wrong syntax: define <name> BresserTemeo <code>".int(@a)
+		if(int(@a) < 3);
+
+  $hash->{CODE}    = $a[2];
+  $hash->{lastMSG} =  "";
+
+  my $name= $hash->{NAME};
+
+  $modules{BresserTemeo}{defptr}{$a[2]} = $hash;
+  $hash->{STATE} = "Defined";
+
+  #AssignIoPort($hash);
+  return undef;
+}
+
+#####################################
+sub
+BresserTemeo_Undef($$)
+{
+  my ($hash, $name) = @_;
+  delete($modules{BresserTemeo}{defptr}{$hash->{CODE}}) if($hash && $hash->{CODE});
+  return undef;
+}
+
+
+#####################################
+sub
+BresserTemeo_Parse($$)
+{
+	my ($iohash,$msg) = @_;
+	my (undef ,$rawData) = split("#",$msg);
+
+	my $prefix = substr $msg, 0, 3;
+
+  my $binvalue = "";
+	my $hexvalue =  substr $msg, 4;
+    #my $bin2 = sprintf( "%b", hex( $hexvalue ) );
+  $binvalue = unpack("B*" ,pack("H*", $hexvalue));
+	if (length($binvalue) != 72)
+    {
+        Log3 $iohash, 4, "BresserTemeo_Parse length error (72 bits expected)!!!";
+        return "";
+    }
+    # Check what Humidity Prefix (*sigh* Bresser!!!) 
+    if ($prefix eq "u44")
+    {
+      $binvalue = "0".$binvalue;
+      Log3 $iohash, 4, "BresserTemeo_Parse Humidity <= 79  Flag";
+    }
+    else
+    {
+      $binvalue = "1".$binvalue;
+      Log3 $iohash, 4, "BresserTemeo_Parse Humidity > 79  Flag";
+    }
+
+    my $checksumOkay = 1;
+	
+	my $hum1 = substr $binvalue, 0, 4;
```

Comment:
my hum1 = hex(substr($hexvalue ,0,1)); würde die nächste Zeile und das umwandeln in bits ersetzen